benchmarking/frameworks/framework_base.py

```diff
@@ -103,6 +109,36 @@ def runBenchmark(self, info, benchmark, platform):
                 platform.delFilesFromPlatform(input_files)
         return output, output_files
 
+    @abc.abstractmethod
+    def composePreprocessCommand(self, preprocess_info):
+        files_db = {'preprocess': {'files': {}}}
+        for f_key in preprocess_info["files"]:
+            f_value = preprocess_info["files"][f_key]
+            if 'location' in f_value:
+                files_db['preprocess']['files'][f_key] = f_value['location']
+            if 'target' in f_value:
+                files_db['preprocess']['files'][f_key] = f_value['target']
+
+        assert 'program' in files_db['preprocess']['files'], "No program in preprocess"
```

Comment:
It might be good not to force a program here. But to run the specified command directly.